### PR TITLE
prow: run gomod.sh as a presubmit

### PR DESCRIPTION
**Pull request by honza**
Depends on https://github.com/metal3-io/baremetal-operator/pull/648

### CODE REVIEW

File: prow/config/config.yaml [hunk 1]
```diff
@@ -259,6 +259,20 @@ presubmits:
           value: "TRUE"
         image: registry.hub.docker.com/library/golang:1.13.7
         imagePullPolicy: Always
+  - name: gomod
```
Comment (dhellmann) [inline prow/config/config.yaml:262]: It looks like this would only apply to the BMO, right? Could you update the commit message to indicate that?

Comment (honza) [reply prow/config/config.yaml:262]: Done

**Comment by russellb** [comment]
/approve

someone else can lgtm when you're ready for it to merge

**Pull request merged**